src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,146 +1,182 @@
 package com.openlattice.shuttle.dates;
 
+import com.openlattice.shuttle.util.Constants;
 import com.openlattice.shuttle.util.Cached;
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.time.LocalDate;
-import java.time.LocalDateTime;
-import java.time.LocalTime;
-import java.time.OffsetDateTime;
+import java.sql.Timestamp;
+import java.time.*;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
+import java.util.Optional;
 import java.util.TimeZone;
 import java.util.concurrent.ExecutionException;
 import java.util.function.BiFunction;
 
 public class JavaDateTimeHelper {
-    private static final Logger logger = LoggerFactory.getLogger(JavaDateTimeHelper.class);
+    private static final Logger logger = LoggerFactory.getLogger( JavaDateTimeHelper.class );
 
-    private final TimeZone      tz;
-    private final String[]      datePatterns;
+    private       ZoneId   zoneId;
+    private       boolean  shouldAddTimezone;
+    private final String[] datePatterns;
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
-        this.tz = tz;
+    public JavaDateTimeHelper( TimeZone tz, String[] datePatterns, Boolean shouldAddTimezone ) {
         this.datePatterns = datePatterns;
+        this.zoneId = tz.toZoneId();
+        this.shouldAddTimezone = shouldAddTimezone;
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    /**
+     * Parse datetime into OffsetDateTime:
+     * - parse into OffsetDateTime as ISO + check whether timezones match
+     * - parse into OffsetDateTime with provided patterns + perform 2-digit-year-fix + check whether timezones match
+     * - parse into LocalDateTime as ISO + add timezone
+     * - parse into LocalDateTime with provided patterns + perform 2-digit-year-fix + add timezone
+     *
+     * @param date - String to parse
+     */
+    public OffsetDateTime parseDateTime( String date ) {
+
+        if ( StringUtils.isBlank( date ) )
+            return null;
+
+        // Try parsing into a OffsetDateTime
+        OffsetDateTime odt = parseFromOffsetDateTime( date );
+         if ( odt != null ) {
+             if ( shouldAddTimezone ) {
+                 TimeZones.checkTimezonesMatch( odt, zoneId );
+             }
+             return odt;
+         }
+
+        // Try parsing into OffsetDateTime with patterns
+        OffsetDateTime odt_p = parseFromPatterns(
+                date,
+                ( toParse, formatter ) -> OffsetDateTime.parse( toParse, formatter ),
+                ( local_odt, datePattern ) -> DecadeChangeHelper
+                        .fixTwoYearPatternOffsetDateTime( local_odt, datePattern ) );
+        if ( odt_p != null ) {
+            if ( shouldAddTimezone )
+                TimeZones.checkTimezonesMatch( odt_p, zoneId );
```

Comment:
I think it helps code readability to be consistent about use of curly brackets with one line children. I personally favor using them just so it's extra clear what code is constrained to the if statement, and I think we usually do that in the rest of the codebase too, so I'd prefer adding them here and below 

ie switching to
```
if (...) {
  ...
}
```
vs
```
if (...)
  ...
```